lib/library.rb

```diff
@@ -0,0 +1,24 @@
+require 'yaml'
+$collection = YAML.load_file('./lib/data.yml')
+
+class Library
+
+  def show_books
+    @available = $collection.select { |book| book[:available]  }
+    @available.each do |book|
+      puts "#{book[:item][:title]} written by #{book[:item][:author]} is available"
```

Comment:
Please avoid using `puts` we only want to return values in each method. This program is not supposed to provide a UI (nor web based or terminal)